### PR TITLE
V8: Fix YSOD when editing media types

### PR DESCRIPTION
**Pull request by kjac**
### Prerequisites

- [x] I have added steps to test this contribution in the description below

If there's an existing issue for this PR then this fixes https://github.com/umbraco/Umbraco-CMS/issues/5894

### Description

As described in #5894, a server error is displayed when you try editing a media type:

image

This PR fixes it:

add-media-type-property

#### Testing this PR

1. Verify that you can add a new property (of any type) to an existing media type.
2. Verify that you can remote the property again.

### CODE REVIEW

File: src/Umbraco.Web/PublishedCache/NuCache/PublishedSnapshotService.cs [hunk 1]
```diff
@@ -1057,7 +1057,10 @@ private void RefreshMediaTypesLocked(List<int> removedIds, List<int> refreshedId
 
                 _mediaStore.UpdateContentTypes(removedIds, typesA, kits);
                 _mediaStore.UpdateContentTypes(CreateContentTypes(PublishedItemType.Media, otherIds.ToArray()).ToArray());
-                _mediaStore.NewContentTypes(CreateContentTypes(PublishedItemType.Media, newIds.ToArray()).ToArray());
+                if(newIds != null && newIds.Any())
```
Comment (Shazwazza) [inline src/Umbraco.Web/PublishedCache/NuCache/PublishedSnapshotService.cs]: `Any()` shouldn't be used here, we are already dealing with a `List<T>` so you can use `Length`. `Any` uses Linq which means it's first going to create an iterator and use that to see if there is anything in there but we already know that with our concrete object.

Also you can see above that we have an extension method to check exactly this which we are already doing: `newIds.IsCollectionEmpty()` so this ext method should be used instead.

Comment (kjac) [reply src/Umbraco.Web/PublishedCache/NuCache/PublishedSnapshotService.cs]: Makes perfect sense, thanks. It's done now 👍

**Comment by Shazwazza** [comment]
All looks good, we actually have these checks in the RefreshContentTypesLocked but not in RefreshMediaTypesLocked. This PR is good, but there's another check missing if you look at RefreshContentTypesLocked . I'll merge and cherry pick to 8.1 and add the additional check. Thanks!

**Comment by vmachacek** [comment]
Hello, can someone please tell me when this will be released?

**Comment by Shazwazza** [comment]
in 8.1.1, most likely shipped in one week.

**Pull request merged**